nick-questions.txt

```diff
@@ -0,0 +1,25 @@
+
+1. Why is the memo field in report so small?  Sourabh mentioned that this was so the whole report could be transmitted via bluetooth.  But in order for contact tracing to be effective, the client needs to download potentially thousands of these reports, i.e. at least one per infected user in the area, to see if they've been exposed.  How can they download these via Bluetooth?  If we think this is an impractical method to download thousands, and instead rely on http, then there's no reason to keep memo small.  Memo can be a rich source of data and I think it should be large enough to answer as-yet-unknown questions that epidemiologists may want to ask.
+
+2. Why does there need to be TWO levels of rotating keys, i.e. both rvk and CEK?  Isn't one level enough?  Consider the following scheme:
+
+cen_0 <- H_cen(rak)
+cen_i <- H_cen(rvk || cen_{i-1})
+
+then a report over the period j1 to j2 is
+
+report <- rvk || cen_{j1} || j1 || j2 || memo
+
+So anyone can re-compute the CENs and compare to their observations just by cen_{j1+1} <- H_cen(rvk || cen_j1)
+
+Doesn't this have all the benefits of the current scheme, without the added complexity of rotating CEK?  You can still rotate rvk as often as you'd like.

```

Comment:
What this loses is forward secrecy; in the current design, someone can reveal their data starting from a particular time, without revealing the prior CENs.